__tests__/textalyze.test.js

```diff
@@ -53,7 +53,7 @@ describe('stringToLetterArray', () => {
 
   test('handles non-string inputs', () => {
     const input = 10;
-    const expectedOutput = [];
+    const expectedOutput = ['1', '0'];
```

Comment:
Because this function now tries to to convert everything to a `String` before doing its work, I'd verify this behavior in the tests.

To do that, I'd create a dummy object with a `toString()` method, e.g.,

```javascript
const object = {toString: () => 'hello');
expect(sanitize(object)).toEqual(['h', 'e', 'l', 'l', 'o']);
```

The test might be something like `converts non-string objects to strings with toString`.